src/service/invoice-service.ts

```diff
@@ -137,39 +164,48 @@ export default class InvoiceService {
       transfer: invoice.transfer ? TransferService.asTransferResponse(invoice.transfer) : undefined,
       description: invoice.description,
       pdf: invoice.pdf ? invoice.pdf.downloadName : undefined,
-      currentState: InvoiceService.asInvoiceStatusResponse(invoice.latestStatus),
+      currentState: InvoiceService.asInvoiceStatusResponse(invoice.invoiceStatus[invoice.invoiceStatus.length - 1]),
       city: invoice.city,
       country: invoice.country,
       postalCode: invoice.postalCode,
       street: invoice.street,
+      totalInclVat: DineroTransformer.Instance.from(totalInclVat).toObject(),
     };
   }
 
   /**
    * Parses an Invoice Object to a InvoiceResponse
    * @param invoice - The Invoice to parse
    */
-  public static asInvoiceResponse(invoice: Invoice)
-    : InvoiceResponse {
+  public async asInvoiceResponse(invoice: Invoice)
+    : Promise<InvoiceResponse> {
+    const customEntries = invoice.invoiceEntries.map(InvoiceService.asInvoiceEntryResponse);
+    const relations: FindOptionsRelations<SubTransactionRow> = {
+      product: {
+        vat: true,
+      },
+    };
+    const subTransactionRows = await this.manager.find(SubTransactionRow, { where: { invoice: { id: invoice.id } }, relations });
+    const transactionEntries = subTransactionRows.map(InvoiceService.subTransactionRowsAsInvoiceEntryResponse);
+
     return {
-      ...this.asBaseInvoiceResponse(invoice),
-      invoiceEntries: invoice.invoiceEntries
-        ? invoice.invoiceEntries.map(this.asInvoiceEntryResponse) : [],
+      ...InvoiceService.asBaseInvoiceResponse(invoice),
+      invoiceEntries: [...customEntries, ...transactionEntries],
     } as InvoiceResponse;
   }
 
-  public static toResponse(invoices: Invoice | Invoice[], entries: boolean): BaseInvoiceResponse | InvoiceResponse | BaseInvoiceResponse[] | InvoiceResponse[] {
+  public async toResponse(invoices: Invoice | Invoice[], entries: boolean): Promise<BaseInvoiceResponse | InvoiceResponse | BaseInvoiceResponse[] | InvoiceResponse[]> {
```

Comment:
I am really not a fan of one method possibly returning four different response types. It is possible to separate all these in their own methods? With so many responses, implicit typing is no longer a thing.